File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        #Solution number 1
        # word=s.split()
        # word.reverse()
        # result=' '.join(word)
        # return result

        #Lets explore solution number 2
        res=[]
        temp=""
        for c in s:
            if c!=" ":
                temp+=c
            elif temp!="":
                res.append(temp)
                temp=""
        if temp!="":
            res.append(temp)

        return " ".join(res[::-1])
        

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String nums[]=s.trim().split("\\s+");

        int i=0;
        int j=nums.length-1;

        while(i<j)
        {
            String temp=nums[i];
            nums[i]=nums[j];
            nums[j]=temp;
            i++;
            j--;
        }
        StringBuilder str=new StringBuilder();
        for(int k=0;k<nums.length;k++)
        {
            str.append(nums[k]);
            str.append(" ");
        }
        str.deleteCharAt(str.length()-1);

        return str.toString();
        
    }
}